staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors_test.go

```diff
@@ -173,3 +173,42 @@ func TestNotSupported(t *testing.T) {
 		t.Errorf("Expected: %s\n, but got: %s\n", expected, notSupported.ErrorBody())
 	}
 }
+
+type stringer struct {
```

Comment:
also add a test case of an object that implements `String()` on the pointer receiver:

```diff
diff --git a/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors_test.go b/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors_test.go
index 4a74c99af3b..bde22eb76d3 100644
--- a/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors_test.go
+++ b/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors_test.go
@@ -181,8 +181,17 @@ func (s stringer) String() string {
 	return "stringer"
 }
 
+type stringerPtr struct {
+	value string
+}
+
+func (s *stringerPtr) String() string {
+	return s.value + "stringerPtr"
+}
+
 func TestErrorBody(t *testing.T) {
 	var nilStringer *stringer = nil
+	var nilStringerPtr *stringerPtr = nil
 	testCases := []struct {
 		name     string
 		err      *Error
@@ -201,6 +210,19 @@ func TestErrorBody(t *testing.T) {
 			err:      Invalid(NewPath("bla"), nilStringer, "details"),
 			expected: "Invalid value: \"null\": details",
 		},
+		{
+			name:     "stringerPtr",
+			err:      Invalid(NewPath("bla"), stringerPtr{}, "details"),
+			expected: `Invalid value: field.stringerPtr{value:""}: details`,
+		}, {
+			name:     "stringerPtr pointer",
+			err:      Invalid(NewPath("bla"), &stringerPtr{}, "details"),
+			expected: "Invalid value: stringerPtr: details",
+		}, {
+			name:     "nil stringerPtr",
+			err:      Invalid(NewPath("bla"), nilStringerPtr, "details"),
+			expected: `Invalid value: "null": details`,
+		},
 	}
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
```